creator-node/src/middlewares.js

```diff
@@ -0,0 +1,130 @@
+const axios = require('axios')
+
+const { sendResponse, errorResponseUnauthorized, errorResponseServerError } = require('./apiHelpers')
+const config = require('./config')
+
+/** Blocks writes if node is not the primary for audiusUser associated with wallet. */
+async function preMiddleware (req, res, next) {
+  if (!req.session || !req.session.wallet) {
+    return sendResponse(req, res, errorResponseUnauthorized('User must be logged in'))
+  }
+
+  let serviceEndpoint
+  try {
+    serviceEndpoint = await _getOwnEndpoint(req)
+  } catch (e) {
+    return sendResponse(req, res, errorResponseServerError(e))
+  }
+
+  let creatorNodeEndpoints
+  try {
+    creatorNodeEndpoints = await _getCreatorNodeEndpoints(req, req.session.wallet)
+  } catch (e) {
+    return sendResponse(req, res, errorResponseServerError(e))
+  }
+  const primaryEndpoint = creatorNodeEndpoints[0]
+
+  // error if self is not primary
+  if (primaryEndpoint && serviceEndpoint !== primaryEndpoint) {
+    return sendResponse(
+      req,
+      res,
+      errorResponseUnauthorized(`This node (${serviceEndpoint}) is not primary for user. Primary is: ${primaryEndpoint}`)
+    )
+  }
+
+  next()
+}
+
+/** Tell all secondaries to sync against self */
+async function postMiddleware (req, res) {
```

Comment:
As discussed offline, I don't think we should since we need more granular control over which routes call/don't call this. I did make it a helper fn instead of a middleware now though